openapi/src/parts/controlled_azure_vm.yaml

```diff
@@ -140,6 +140,10 @@ components:
           description: Supply base64 encoded custom data for vm. https://docs.microsoft.com/en-us/azure/virtual-machines/custom-data
           type: string
           format: base64
+        priority:
+          description: Priority of the VM (Regular or Spot). Defaults to Regular.
+          type: string
+          enum: [ 'REGULAR', 'SPOT' ]
```

Comment:
What happens to the WSM resource when a spot instance disappears?

Reply:
In WSM: nothing. The resource continues to exist.

We are using the default `deallocate` eviction policy though, which means the Azure VM does not completely disappear: it moves into a stopped-deallocated state, and can be redeployed (or deleted) later. So there is still an Azure resource even if eviction happens.